src/adapt.rs

```diff
@@ -0,0 +1,481 @@
+//! An API for plugging in adapters.
+
+#![allow(dead_code)] // Implementation in progress, code isn't called yet.
+
+use transact::{InsertInMap, InsertInSet};
```

Comment:
We use the transact:: { InsertInMap, InsertInSet } style everywhere else in foxbox. I don't know if the "no space" version is the commonly used style in the rust community, but I find it less readable. In any case, we need to choose one style and stick to it.